python/foglamp/services/core/api/task.py

```diff
@@ -215,8 +217,8 @@
                 for k, v in config.items():
                     await config_mgr.set_category_item_value_entry(name, k, v['value'])
         except Exception as ex:
-            await revert_configuration(storage, name)  # Revert configuration entry
-            await revert_parent_child_configuration(storage, name)
+            await delete_configuration(storage, name)  # Revert configuration entry
+            await delete_parent_child_configuration(storage, name)
```

Comment:
We should also recursively delete child categories of name. E.g. the Advanced children of a south service.

Reply:
This is for north instance. the feedback on south endpoint is handled in https://github.com/foglamp/FogLAMP/pull/1198/files#diff-e21909c861b284ca7159d051a8d325d1R289